test/serializers/attribute_test.rb

```diff
@@ -71,6 +71,21 @@ def id
 
         assert_equal('custom', hash[:blog][:id])
       end
+
+      PostWithVirtualAttribute = Class.new(::Model)
+      class PostWithVirtualAttributeSerializer < ActiveModel::Serializer
+        attribute :name do
+          "#{object.first_name} #{object.last_name}"
+        end
+      end
```

Comment:
I mean, right now it is `self.class._fragmented ? self.class._fragmented.public_send(key) : send(key)` but it could me `self.class._fragmented ? self.class._fragmented.public_send(key) : object.read_attribute_for_serialization(key)`